Add com.scanoss.sbom-workbench

perezale opened the pull request:
The SBOM Workbench is a graphical user interface to scan and audit source code using SCANOSS API.

Auditing your source code for license compliance has never been easier. Simply scan your source code directory to find and identify open source components. Generate your SPDX-Lite software bill of materials (SBOM) with the press of a button.

### Please confirm your submission meets all the criteria

<!-- Please replace each `[ ]` by `[X]` when the step is complete -->

- [X] Please describe your application briefly.
- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I have [built][build] and tested the submission locally.
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

hfiguiere commented on com.scanoss.sbom-workbench.yml:
```diff
@@ -0,0 +1,44 @@
+      - install -Dm644 -t ${FLATPAK_DEST}/share/metainfo com.scanoss.sbom-workbench.metainfo.xml
+    sources:
+      - type: file
+        url: https://github.com/scanoss/sbom-workbench/releases/download/v1.12.4/sbom-workbench-1.12.4-linux-x86_64.AppImage
```
why not building from source? that would bring aarch64 support.

perezale replied:
Hi Hubert!  Thank you for taking a look at our request.

We've asked for instructions at flathub@lists.freedesktop.org and we received help from Bartłomiej. 

He took a look at our project and although he mentioned the possibility of building from sources in your infrastructure, he thinks it will be painful and he didn't recommend this approach for our project. He presented us with an alternative to repackage our AppImage package.

We definitely intend to add this to increase visibility and will add it our backlog

As I informed Bartłomiej, there's a Spanish KDE developers event this weekend, and we would like to showcase the SBOM Workbench available at Flathub (or nearby). It would mean a lot to us to have this opportunity.

Do you think we could plan to support aarch64 in a future release?

Also, any other suggestion is welcomed. Thank you very much for your assistance.

bbhtt commented on run.sh:
```diff
@@ -0,0 +1,31 @@
+        echo "To disable, remove the --socket=wayland permission."
+        EXTRA_FLAGS+=(
+            "--enable-features=UseOzonePlatform,WaylandWindowDecorations"            
+            "--ozone-platform=wayland"
+            "--ozone-platform-hint=auto"            
```
You don't need both these flags. The last one is the latest one, the previous one is deprecated. You should check what flag your Electron version supports.

bbhtt commented on run.sh:
```diff
@@ -0,0 +1,31 @@
+declare -a EXTRA_FLAGS=()
+
+# Display Socket
+if [[ $XDG_SESSION_TYPE == "wayland" ]]
+then
+    WAYLAND_SOCKET=${WAYLAND_DISPLAY:-"wayland-0"}
+
+    if [[ -e "$XDG_RUNTIME_DIR/${WAYLAND_SOCKET}" ]]
```
Combine this as if condition1 && condition2

bbhtt commented on run.sh:
```diff
@@ -0,0 +1,31 @@
+fi
+
+cd /app/lib/sbom-workbench
+export TMPDIR="$XDG_CACHE_HOME"
```
This is persistent on disk and not tmpfs just to note.

bbhtt commented on com.scanoss.sbom-workbench.yml:
```diff
@@ -0,0 +1,44 @@
+  - "--persist=.scanoss"
+  - "--persist=scanoss-workspace"
+  - "--persist=.pki"
+  - "--filesystem=/tmp:rw"
```
If you set TMPDIR you don't need this.

bbhtt commented on com.scanoss.sbom-workbench.metainfo.xml:
```diff
@@ -0,0 +1,37 @@
+<component type="desktop-application">
+  <id>com.scanoss.sbom-workbench</id>
+  <name>SCANOSS SBOM Workbench</name>
+  <summary>Graphical user interface to scan and audit your source code.</summary>
```
```suggestion
  <summary>Graphical user interface to scan and audit your source code</summary>
```

bbhtt commented on com.scanoss.sbom-workbench.metainfo.xml:
This file should be upstream

bbhtt commented on com.scanoss.sbom-workbench.png:
Icon should be upstream

perezale commented:
Hi team! Thanks for reviewing the submission so quickly. I'll try my best to send the updates you require. 

Last update:
Made metainfo and icon available upstream (and removed them from the repo, is that ok?)
Removed unused permission on /tmp
Unified conditionals on run.sh

bbhtt commented:
bot, build com.scanoss.sbom-workbench

perezale commented:
> Build [122364 failed](https://buildbot.flathub.org/#/builders/6/builds/122364)

Oh! hi! I've checked our load balancer and the resource download got rate limited, too fast! :)  I've added an exception to allow bursts of 10 requests to exceed our defined rate limit. Let me know if that should work for the build process.

bbhtt commented:
bot, build com.scanoss.sbom-workbench

Shipping the icon inside the appimage would be better.

bbhtt commented:
metadata license is not a spdx identifier.

bbhtt commented:
Also GPL isn't allowed for metadata license, use a permissive one like FSFAP, CC0-1.0 etc.

And the categories `Operations` and `SoftwareCompositionAnalysis` are invalid. Remove them.

perezale commented:
> metadata license is not a spdx identifier.

> Also GPL isn't allowed for metadata license, use a permissive one like FSFAP, CC0-1.0 etc.
> 
> And the categories `Operations` and `SoftwareCompositionAnalysis` are invalid. Remove them.

Thanks! it's now updated!

bbhtt commented:
bot, build com.scanoss.sbom-workbench

bbhtt commented:
bot, build com.scanoss.sbom-workbench

bbhtt commented:
`<launchable type="desktop-id">com.scanoss.scanoss-workbench.desktop</launchable>` is wrong it should be `com.scanoss.sbom-workbench.desktop`

perezale commented:
> `<launchable type="desktop-id">com.scanoss.scanoss-workbench.desktop</launchable>` is wrong it should be `com.scanoss.sbom-workbench.desktop`

alright! I'll check everything works locally before and lyk

bbhtt commented:
bot, build com.scanoss.sbom-workbench
